Tools/WebKitTestRunner/ios/UIScriptControllerIOS.mm

```diff
@@ -106,6 +106,46 @@ static BOOL returnNo()
     };
 }
 
+static RetainPtr<NSDictionary> toNSDictionary(CGPathRef path)
+{
+    auto pathElementTypeToString = [](CGPathElementType type) {
+        switch (type) {
+        case kCGPathElementMoveToPoint:
+            return @"MoveToPoint";
+
+        case kCGPathElementAddLineToPoint:
+            return @"AddLineToPoint";
+
+        case kCGPathElementAddQuadCurveToPoint:
+            return @"AddQuadCurveToPoint";
+
+        case kCGPathElementAddCurveToPoint:
+            return @"AddCurveToPoint";
+
+        case kCGPathElementCloseSubpath:
+            return @"CloseSubpath";
+
+        default:
+            return @"Unknown";
+        }
+    };
+
+    auto attributes = adoptNS([[NSMutableDictionary alloc] init]);
```

Comment:
This really should have been an array of key-value pairs, since it's very possible to have the same path element type show up multiple times in a given path.